server/src/internalClusterTest/java/org/elasticsearch/action/search/CCSPointInTimeIT.java

```diff
@@ -71,6 +105,112 @@ public void testBasic() {
+            SearchResponse.Cluster remoteCluster = remoteClusterRef.get();
+            assertThat(remoteCluster.getTotalShards(), equalTo(1));
+            assertThat(remoteCluster.getSuccessfulShards(), equalTo(1));
+            assertThat(remoteCluster.getFailedShards(), equalTo(0));
+            assertThat(remoteCluster.getFailures().size(), equalTo(0));
+            assertThat(remoteCluster.getStatus(), equalTo(SearchResponse.Cluster.Status.SUCCESSFUL));
+            assertThat(remoteCluster.getTook().millis(), greaterThanOrEqualTo(0L));
+            assertFalse(remoteCluster.isTimedOut());
+
+        } finally {
+            closePointInTime(pitId);
+        }
+    }
+
+    public void testFailuresOnOneShardsWithPointInTime() throws ExecutionException, InterruptedException {
+        final Client localClient = client(LOCAL_CLUSTER);
+        final Client remoteClient = client(REMOTE_CLUSTER);
+        int localNumDocs = randomIntBetween(10, 50);
+        int numShards = randomIntBetween(2, 4);
+        Settings clusterSettings = indexSettings(numShards, randomIntBetween(0, 1)).build();
+        assertAcked(localClient.admin().indices().prepareCreate("local_test").setSettings(clusterSettings));
+        indexDocs(localClient, "local_test", localNumDocs);
+
+        int remoteNumDocs = randomIntBetween(10, 50);
+        assertAcked(remoteClient.admin().indices().prepareCreate("remote_test").setSettings(clusterSettings));
+        indexDocs(remoteClient, "remote_test", remoteNumDocs);
+        boolean includeLocalIndex = randomBoolean();
+        List<String> indices = new ArrayList<>();
+        if (includeLocalIndex) {
+            indices.add(randomFrom("*", "local_*", "local_test"));
+        }
+        indices.add(randomFrom("*:*", "remote_cluster:*", "remote_cluster:remote_test"));
+        String pitId = openPointInTime(indices.toArray(new String[0]), TimeValue.timeValueMinutes(2));
+        try {
+            if (randomBoolean()) {
+                localClient.prepareIndex("local_test").setId("local_new").setSource().get();
+                localClient.admin().indices().prepareRefresh().get();
+            }
+            if (randomBoolean()) {
+                remoteClient.prepareIndex("remote_test").setId("remote_new").setSource().get();
+                remoteClient.admin().indices().prepareRefresh().get();
+            }
+            PlainActionFuture<SearchResponse> queryFuture = new PlainActionFuture<>();
+
+            // shardId 0 means to throw the Exception only on shard 0; all others should work
+            ThrowingQueryBuilder queryBuilder = new ThrowingQueryBuilder(randomLong(), new IllegalStateException("index corrupted"), 0);
+            SearchRequest searchRequest = new SearchRequest();
+            searchRequest.source(new SearchSourceBuilder().query(queryBuilder).size(10).pointInTimeBuilder(new PointInTimeBuilder(pitId)));
+            client(LOCAL_CLUSTER).search(searchRequest, queryFuture);
```

Comment:
You can just use:

```java
            SearchResponse searchResponse = client(LOCAL_CLUSTER).search(searchRequest).get();
```

save a couple lines for the explicit future and such.

Reply:
Thanks. I also modified several of the tests in CrossClusterSearchIT this way.